react/features/video-layout/middleware.web.js

```diff
@@ -99,3 +108,76 @@ MiddlewareRegistry.register(store => next => action => {
 
     return result;
 });
+
+/**
+ * Middleware which listens for actions and performs updates related to the
+ * auto-pin screen share feature.
+ *
+ * @param {Store} store - The redux store.
+ * @returns {Function}
+ */
+MiddlewareRegistry.register(store => next => action => {
```

Comment:
Why do you register second middleware?

Reply:
Actually you are correct, there shouldn't be any performance implications. I got confused...

But still, I personally would prefer the function option. But if you are OK with the 2 middlewares, I'll agree with it!